src/server/service/kv.rs

```diff
@@ -1550,6 +1561,50 @@ fn future_batch_get<E: Engine, L: LockManager, F: KvFormat>(
     }
 }
 
+fn future_buffer_batch_get<E: Engine, L: LockManager, F: KvFormat>(
+    storage: &Storage<E, L, F>,
+    mut req: BufferBatchGetRequest,
+) -> impl Future<Output = ServerResult<BufferBatchGetResponse>> {
+    let tracker = GLOBAL_TRACKERS.insert(Tracker::new(RequestInfo::new(
+        req.get_context(),
+        RequestType::KvBufferBatchGet,
+        req.get_version(),
+    )));
+    set_tls_tracker_token(tracker);
+    let start = Instant::now();
+    let keys = req.get_keys().iter().map(|x| Key::from_raw(x)).collect();
+    let v = storage.buffer_batch_get(req.take_context(), keys, req.get_version().into());
+
+    async move {
+        let v = v.await;
+        let duration = start.saturating_elapsed();
+        let mut resp = BufferBatchGetResponse::default();
+        if let Some(err) = extract_region_error(&v) {
+            resp.set_region_error(err);
+        } else {
+            match v {
+                Ok((kv_res, stats)) => {
+                    let pairs = map_kv_pairs(kv_res);
+                    let exec_detail_v2 = resp.mut_exec_details_v2();
+                    let scan_detail_v2 = exec_detail_v2.mut_scan_detail_v2();
+                    stats.stats.write_scan_detail(scan_detail_v2);
+                    GLOBAL_TRACKERS.with_tracker(tracker, |tracker| {
+                        tracker.write_scan_detail(scan_detail_v2);
+                    });
+                    set_time_detail(exec_detail_v2, duration, &stats.latency_stats);
```

Comment:
There is a newly introduced grpc duration(https://github.com/tikv/tikv/pull/16447), need to make sure if it's recorded.

Reply:
Thanks for the reminder. It looks like a problem that our code structure and git cannot detect such problem.